### PR TITLE
hacker news with 0.15.0 asset APIs

### PR DESCRIPTION
**Pull request by sryza**

### CODE REVIEW

File: examples/hacker_news_assets/hacker_news_assets/activity_analytics/__init__.py [hunk 1]
```diff
@@ -1,68 +1,32 @@
...
 
-from dagster import AssetGroup
+from dagster import assets_from_package_module
 from dagster.utils import file_relative_path
 
 from . import assets
 
 DBT_PROJECT_DIR = file_relative_path(__file__, "hacker_news_dbt")
 DBT_PROFILES_DIR = DBT_PROJECT_DIR + "/config"
-dbt_staging_resource = dbt_cli_resource.configured(
```
Comment (sryza) [inline examples/hacker_news_assets/hacker_news_assets/activity_analytics/__init__.py:16]: Looks like this actually wasn't being used.

File: python_modules/dagster/dagster/core/asset_defs/asset_group.py [hunk 2]
```diff
@@ -648,3 +648,39 @@ def _validate_resource_reqs_for_asset_group(
                 "AssetGroup is missing required resource keys for resource '"
                 f"{resource_key}'. Missing resource keys: {missing_resource_keys}"
             )
+
+
+def assets_from_package_module(
```
Comment (sryza) [inline python_modules/dagster/dagster/core/asset_defs/asset_group.py:653]: placeholder: not fully implemented

**Review by OwenKephart — APPROVED** [review]
nice, this feels a lot better

there's definitely a tradeoff in terms of how much you need to specify in the asset decorator (sometimes it's more, sometimes it's less), but I think in the new state the information that you're specifying feels more relevant to the specific asset (whereas before I often felt like I was just setting random parameters to make things link together)

File: examples/hacker_news_assets/hacker_news_assets/__init__.py [hunk 3]
```diff
@@ -1,48 +1,45 @@
...
 
 
 @repository
-def local():
-    return [
-        *core_definitions_local,
-        *recommender_definitions_local,
-        *activity_analytics_definitions_local,
-    ]
+def repo():
```
[the remark this answers is not in the document]
Comment (OwenKephart) [reply examples/hacker_news_assets/hacker_news_assets/__init__.py:34]: any changes other than replacing the three repositories with this?

I'm on board with this approach, but I think reorganizing this function slightly might make it easier for someone browsing this code to understand what's going on. I'm imagining something like:

```python

all_resources = {"prod": RESOURCES_PROD, ...}

@repository
def repo():
    deployment_name = os.environ.get("DAGSTER_DEPLOYMENT", "local")

    resource_defs = all_resources[deployment_name]
    definitions = with_resources(all_assets, resource_defs) + all_jobs

    if deployment_name in ["prod", "staging"]:
        definitions.append(make_slack_on_failure_sensor(base_url="my_dagit_url"))

    return definitions
```

[the remark this answers is not in the document]
Comment (sryza) [reply examples/hacker_news_assets/hacker_news_assets/__init__.py:40]: no other changes than that

I reorganized the function in the way you described

**Pull request merged**